From da39d3af836c5563b88b0069fcc56d5209892777 Mon Sep 17 00:00:00 2001
From: Jos van den Oever <jos@vandenoever.info>
Date: Thu, 28 Jul 2011 16:57:45 +0200
Subject: [PATCH] Fix crash on assert when file grows while reading it. Bug
 reported by Michael Jansen.

Source: upstream, libstreams.git
Upstream: yes

---
 lib/dataeventinputstream.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/libstreams/lib/dataeventinputstream.cpp b/libstreams/lib/dataeventinputstream.cpp
index a596706..8eac0a9 100644
--- a/libstreams/lib/dataeventinputstream.cpp
+++ b/libstreams/lib/dataeventinputstream.cpp
@@ -44,6 +44,11 @@ DataEventInputStream::read(const char*& start, int32_t min, int32_t max) {
         return -2;
     }
     if (nread > 0) {
+        // ignore bytes that might have been added since the file size was
+        // determined
+        if (m_size != -1 && m_position + nread > m_size) {
+            nread = m_size - m_position;
+        }
         m_position += nread;
         // value of -1 for totalread means data should not be reported anymore
         if (totalread != -1 && totalread < m_position) {
-- 
1.7.6.1

